flopy/modpath/mp7particledata.py

```diff
@@ -778,10 +786,16 @@ def write(self, f=None):
 )
 
 
-def get_extent(grid, k=None, i=None, j=None, nn=None, localz=False) -> Extent:
+def get_extent(
+    grid, k=None, i=None, j=None, nn=None, localz=False, global_xy=False
+) -> Extent:
     # get cell coords and span in each dimension
     if not (k is None or i is None or j is None):
         verts = grid.get_cell_vertices(i, j)
+        if not global_xy and (
+            grid.xoffset != 0.0 or grid.yoffset != 0.0 or grid.angrot != 0.0
```

Comment:
this could use `grid._has_ref_coordinates` &mdash; which incidentally I'd vote for making "public" at some point by dropping the underscore